std-bits/base/src/main/java/org/enso/base/enso_cloud/EnsoSecretReader.java

```diff
@@ -0,0 +1,67 @@
+package org.enso.base.enso_cloud;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.HashMap;
+import java.util.Map;
+
+/** * Internal class to read secrets from the Enso Cloud. */
+class EnsoSecretReader {
+  private static Map<String, String> secrets = new HashMap<>();
+
+  static void flushCache() {
+    secrets.clear();
+  }
+
+  /**
+   * * Reads a secret from the Enso Cloud.
+   *
+   * @param secretId the ID of the secret to read.
+   * @return the secret value.
+   */
+  static String readSecret(String secretId) {
+    if (secrets.containsKey(secretId)) {
+      return secrets.get(secretId);
+    }
+
+    var apiUri = AuthenticationProvider.getAPIRootURI() + "/secrets/" + secretId;
+    var client = HttpClient.newBuilder().followRedirects(HttpClient.Redirect.ALWAYS).build();
+    var request =
+        HttpRequest.newBuilder()
+            .uri(URI.create(apiUri))
+            .header("Authorization", "Bearer " + AuthenticationProvider.getToken())
+            .GET()
+            .build();
+
+    HttpResponse<String> response;
+
+    try {
+      response = client.send(request, HttpResponse.BodyHandlers.ofString());
+    } catch (IOException | InterruptedException e) {
+      throw new IllegalArgumentException("Unable to read secret.");
+    }
+
+    if (response.statusCode() != 200) {
+      throw new IllegalArgumentException("Unable to read secret.");
+    }
+
+    var secretJSON = response.body();
+    var secretValue = readValueFromString(secretJSON);
+    secrets.put(secretId, secretValue);
+    return secretValue;
+  }
+
+  private static String readValueFromString(String json) {
+    return json.substring(1, json.length() - 1)
+        .replace("\\\"", "\"")
+        .replace("\\n", "\n")
+        .replace("\\t", "\t")
+        .replace("\\r", "\r")
+        .replace("\\b", "\b")
+        .replace("\\f", "\f")
+        .replace("\\\\", "\\");
+  }
```

Comment:
However, merely reordering the `replace`s will probably not help either, because if we have e.g. `"\\n"` and _first_ replace `\\` by `\`, then later look for `\n` it will match and in such case the problematic string would become just a newline character, which is still a wrong interpretation of that JSON value.

Reply:
indeed this was a quick hack but not intended to stay

Reply:
Replaces with `translateEscapes()` which I failed to find when I was putting this quick workaround in.